docs: update docs due v0.20.0 (#1110)
deckhouse/virtualization: docs/USER_GUIDE.md

### Placement of VMs by nodes

The following methods can be used to manage the placement of virtual machines (placement parameters) across nodes:

- Simple label selection (`nodeSelector`) — the basic method for selecting nodes with specified labels.
- Preferred selection (`Affinity`):
- `nodeAffinity` — specifies priority nodes for placement.
  - `virtualMachineAndPodAffinity` — ensures co-location of VMs and containers.
- Co-location avoidance (`AntiAffinity`):

> - `virtualMachineAndPodAntiAffinity` — prevents VMs and containers from being placed on the same node.

Reviewer: ```suggestion
- `virtualMachineAndPodAntiAffinity` — defines workload rules for VMs or containers to be placed on the same node.
```

- Creating consistent placement rules.
- Checking the compatibility of rules before applying them.
- Consider the types of conditions:
- Strict (`requiredDuringSchedulingIgnoredDuringExecution`) — require strict compliance.
- Soft (`preferredDuringSchedulingIgnoredDuringExecution`) — allow partial compliance.
- Use combinations of labels instead of single restrictions. For example, instead of required for a single label (env=prod), use several preferred conditions.
- Consider the order in which interdependent VMs are launched. When using Affinity between VMs (for example, the backend depends on the database), launch the VMs referenced by the rules first to avoid lockouts.
- Plan backup nodes for critical workloads. For VMs with strict requirements (e.g., AntiAffinity), provide backup nodes to avoid downtime in case of failure.
- Consider existing  `taints` on nodes.

{{< alert level="info" >}}
When changing placement parameters:
- If the current location of the VM meets the new requirements, it remains on the current node.
- If the requirements are violated:

> - In paid editions: The VM is automatically moved to a suitable node using live migration.

Reviewer: ```suggestion
- In commercial editions: The VM is automatically moved to a suitable node using live migration.
```
